source/common/conn_pool/conn_pool_base.cc

```diff
@@ -33,11 +33,44 @@ void ConnPoolImplBase::destructAllConnections() {
   dispatcher_.clearDeferredDeleteList();
 }
 
-void ConnPoolImplBase::tryCreateNewConnection() {
-  if (pending_requests_.size() <= connecting_request_capacity_) {
-    // There are already enough CONNECTING connections for the number
-    // of queued requests.
-    return;
+bool ConnPoolImplBase::shouldCreateNewConnection() const {
+  // The number of requests we want to be provisioned for is the number of
+  // pending and active requests times the prefetch ratio.
+  // The number of requests we are (theoretically) provisioned for is the
+  // connecting request capacity plus the number of active requests.
+  //
+  // If prefetch ratio is not set, it defaults to 1, and this simplifies to the
+  // legacy value of pending_requests_.size() > connecting_request_capacity_
+  return (pending_requests_.size() + num_active_requests_) * prefetchRatio() >
+         (connecting_request_capacity_ + num_active_requests_);
+}
+
+float ConnPoolImplBase::prefetchRatio() const {
+  if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.allow_prefetch")) {
```

Comment:
I don't feel strongly about it but you could probably get away with latching this value on the conn pool vs. looking it up every time.

Reply:
I'm mildly inclined to leave it where it is, because otherwise flipping the runtime flag won't fix things under load, unless lifetime requests per connection is set.